library/psa_crypto.c

```diff
@@ -3837,7 +3844,7 @@ static psa_status_t psa_cipher_setup( psa_cipher_operation_t *operation,
     operation->key_set = 1;
     operation->block_size = ( PSA_ALG_IS_STREAM_CIPHER( alg ) ? 1 :
                               PSA_BLOCK_CIPHER_BLOCK_SIZE( slot->attr.type ) );
-    if( alg & PSA_ALG_CIPHER_FROM_BLOCK_FLAG )
+    if( alg & PSA_ALG_CIPHER_FROM_BLOCK_FLAG && alg != PSA_ALG_ECB_NO_PADDING )
```

Comment:
Suggestion: Add brackets for clarity.
```suggestion
    if( ( alg & PSA_ALG_CIPHER_FROM_BLOCK_FLAG ) != 0 && 
        alg != PSA_ALG_ECB_NO_PADDING )
```